richtextfx/src/integrationTest/java/org/fxmisc/richtext/style/StylingTests.java

```diff
@@ -0,0 +1,160 @@
+    }
+
+    /**
+     * @return A list of nodes which render the underlines for the current text.
+     */
+    private List<Path> getUnderlinePaths() {
+        // get the ParagraphBox (protected subclass of Region) 
+        Region paragraphBox = getParagraphBox(0);
+        
+        // get the ParagraphText (protected subclass of TextFlow) 
+        TextFlow tf = (TextFlow) paragraphBox.getChildrenUnmodifiable().get(0);
+
+        List<Path> result = new ArrayList<>();
+        tf.getChildrenUnmodifiable().filtered(n -> n instanceof Path).forEach(n -> result.add((Path) n));
+        return result.subList(2, result.size());
+    }
+
+
+    @Test
+    public void simpleStyling() {
+        // setup
+        interact(() -> {
+            area.replaceText(firstWord + styledWord1 + remainingLine);
+        });
+
+        // expected: one text node which contains the complete text
+        List<Text> textNodes = getTextNodes();
+        assertEquals(1, textNodes.size());
+
+        // Set word "World" to bold
+        interact(() -> {
+            area.setStyle(firstWord.length(), firstWord.length() + styledWord1.length(), "-fx-font-weight: bold;");
+        });
+
+        // expected: three text nodes
+        textNodes = getTextNodes();
+        assertEquals(3, textNodes.size());
+
+        Text first = textNodes.get(0);
+        assertEquals("Hello ", first.getText());
+        assertEquals("Regular", first.getFont().getStyle());
+
+        Text second = textNodes.get(1);
+        assertEquals("World", second.getText());
+        assertEquals("Bold", second.getFont().getStyle());
+
+        Text third = textNodes.get(2);
+        assertEquals(" and Moon", third.getText());
+        assertEquals("Regular", third.getFont().getStyle());
+    }
+
+
+    @Test
+    public void underlineStyling() {
+        // setup
+        interact(() -> {
+            area.replaceText(firstWord + styledWord1 + moreText + styledWord2 + remainingLine);
+            //               "Hello World and also the Sun and Moon"
+        });
+
+        // expected: one text node which contains the complete text
+        List<Text> textNodes = getTextNodes();
+        assertEquals(1, textNodes.size());
+        assertEquals(firstWord + styledWord1 + moreText + styledWord2 + remainingLine, 
+                     textNodes.get(0).getText());
+
+        interact(() -> {
+            final int start1 = firstWord.length();
+            final int end1 = start1 + styledWord1.length();
+            area.setStyle(start1, end1,
+                          "-rtfx-underline-color: red; -rtfx-underline-dash-array: 2 2; -rtfx-underline-width: 1; -rtfx-underline-cap: butt;");
+
+            final int start2 = end1 + moreText.length();
+            final int end2 = start2 + styledWord2.length();
+            area.setStyle(start2, end2,
+                          "-rtfx-underline-color: red; -rtfx-underline-dash-array: 2 2; -rtfx-underline-width: 1; -rtfx-underline-cap: butt;");
```

Comment:
Could this be declared once in a variable and then just pass that variable in to both `area.setStyle()` calls?